core/src/test/scala/com.mediative.sparrow/DataFrameReaderTest.scala

```diff
@@ -153,12 +168,58 @@ class DataFrameReaderTest extends FreeSpec with BeforeAndAfterAll {
     }
 
     def testFailure[T: RowConverter: ClassTag](json: Array[String], expected: NonEmptyList[String]) = {
-      val sqlContext = new SQLContext(sc)
       val df = sqlContext.jsonRDD(sc.parallelize(json))
 
       assert(toRDD[T](df) == expected.failure)
     }
 
+    // To get DataFrame#toRDD usage.
+    import com.mediative.sparrow.syntax.df._
+
+    "round-trip TestToRdd1 from RDD to DataFrame back to RDD" in {
+      import TestCaseClasses.TestToRdd1
+
+      val expected = TestToRdd1(1, "a")
+      val df = sqlContext.createDataFrame(sc.parallelize(List(expected)))
+
+      assert(df.toRDD[TestToRdd1].toOption.get.first == expected)
+    }
+
+    "round-trip TestToRdd2 from RDD to DataFrame back to RDD" - {
+      "when containing some value" in {
+        import TestCaseClasses.TestToRdd2
+
+        val expected = TestToRdd2(1, Option(1))
+        val df =
+          sqlContext.createDataFrame(sc.parallelize(List(expected)))
+
+        assert(df.toRDD[TestToRdd2].toOption.get.first == expected)
+      }
+
+      "when containing no value" in {
+        import TestCaseClasses.TestToRdd2
+
+        val expected = TestToRdd2(1, Option.empty)
+        val df =
+          sqlContext.createDataFrame(sc.parallelize(List(expected)))
+
+        assert(df.toRDD[TestToRdd2].toOption.get.first == expected)
+      }
+    }
+
+    "round-trip TestToRdd3 from RDD to DataFrame back to RDD" in {
```

Comment:
@yawaramin minor comment, can we give better names than `TestToRdd3`. it's not so important when naming the case classes, as in the description of the tests (which should be humanly understandable)